nautobot/core/nautobot_app/__init__.py

```diff
@@ -0,0 +1,169 @@
+from django.apps import AppConfig
+from django.urls import reverse
+from operator import getitem
+from collections import OrderedDict
+
+from nautobot.extras.choices import CustomLinkButtonClassChoices
+from nautobot.extras.plugins.utils import import_object
+from nautobot.extras.registry import registry
+
+
+class NautobotConfig(AppConfig):
+    menu_tabs = "navigation.menu_tabs"
+
+    def ready(self):
+        if self.__module__.endswith(".apps"):
+            self.__module__ = self.__module__[:-5]
+        menu_items = import_object(f"{self.__module__}.{self.menu_tabs}")
+        if menu_items is not None:
+            register_menu_items(menu_items)
+
+
+def register_menu_items(class_list):
+    for nav_tab in class_list:
+        registry["nav_menu"]["tabs"][nav_tab.name] = {
+            "weight": nav_tab.weight,
+            "groups": registry["nav_menu"]["tabs"].get(nav_tab.name, {}).get("groups", {}),
+            "permissions": [],
+        }
+
+        tab_perms = []
+        registry_groups = registry["nav_menu"]["tabs"][nav_tab.name]["groups"]
+        for group in nav_tab.groups:
+            registry_groups[group.name] = {
+                "weight": group.weight,
+                "items": registry_groups.get(group.name, {}).get("items", {}),
+                "permissions": [],
+            }
+
+            group_perms = []
+            for item in group.items:
+                registry_groups[group.name]["items"][item.link] = {
```

Comment:
Need a check here against the (admittedly unlikely) case where we try to re-register an already registered link.

Alternatively, why not have `"items"` be a list instead of a dictionary? That way you can have multiple items with the same link coexist if that's what's really desired.